server/src/main/java/org/eclipse/hono/server/HonoServer.java

```diff
@@ -109,25 +114,124 @@ public void start(final Future<Void> startupHandler) {
         }, startupHandler);
     }
 
+    private ProtonServer createProtonServer(ProtonServerOptions options) {
+        return ProtonServer.create(vertx, options)
+                .saslAuthenticatorFactory(saslAuthenticatorFactory)
+                .connectHandler(this::handleRemoteConnectionOpen);
+    }
+
+    /**
+     * Either one secure, one insecure or one secure AND insecure ports are supported in Hono.
+     * The secure port should stick to the IANA default 5671, otherwise a warning will be issued.
+     * The insecure port should stick to the IANA default 5672, otherwise a warning will be issued.
+     *
+     * @param portConfigurationTracker The future to continue or fail (in case of configuration errors).
+     * @throws NullPointerException if portConfigurationTracker is null.
+     */
+    void doPortConfigurations(Future<Void> portConfigurationTracker) {
+        Objects.requireNonNull(portConfigurationTracker);
+
+        // secure port first
+        doSecurePortConfiguration();
+        doInsecurePortConfiguration();
+
+        if (!openSecurePort && !openInsecurePort) {
+            final String message = "cannot start up HonoServer - no listening ports found.";
+            LOG.error(message);
+            portConfigurationTracker.fail(message);
+        } else if (openSecurePort && openInsecurePort && honoConfig.getPort() == honoConfig.getInsecurePort()) {
+            final String message = String.format("cannot start up HonoServer - secure and insecure ports configured to same port number %s.",
+                    honoConfig.getPort());
+            LOG.error(message);
+            portConfigurationTracker.fail(message);
+        } else {
+            portConfigurationTracker.complete();
+        }
+    }
+
+    private void doSecurePortConfiguration() {
+        if (StringUtils.isEmpty(honoConfig.getKeyStorePath())) {
+            if (honoConfig.getPort() >= 0) {
+                LOG.warn("AMQP 1.0 secure port number configured, but no keyStorePath set. No secure port will be opened!");
+            }
+            return;
+        }
+
+        if (honoConfig.getPort() == HonoConfigProperties.PORT_UNCONFIGURED) {
+            honoConfig.setPort(Constants.PORT_AMQP_SECURE);
+            LOG.info("HonoServer found secure port number unconfigured, using IANA standard port {}", Constants.PORT_AMQP_SECURE);
+        } else if (honoConfig.getPort() == 0) {
+            LOG.info("HonoServer found secure port number configured for ephemeral port selection (port chosen automatically).");
+        }
+        openSecurePort = true;
+    }
+
+    private void doInsecurePortConfiguration() {
+        if (!honoConfig.isAllowInsecure())
+            return;
+
+        if (honoConfig.getInsecurePort() == 0) {
+            LOG.info("HonoServer found insecure port number configured for ephemeral port selection (port chosen automatically).");
+        } else if (honoConfig.getInsecurePort() == HonoConfigProperties.PORT_UNCONFIGURED) {
```

Comment:
To not let the pure POJO class HonoConfigProperties depend on a logging framework, I will only move the currently unclean handling of port setting to the property class (avoiding to reconfigure the class from the outside).
I guess this is what you addressed?